src/wp-includes/option.php

```diff
@@ -2924,3 +3028,14 @@ function filter_default_option( $default_value, $option, $passed_default ) {
 
 	return $registered[ $option ]['default'];
 }
+
+/**
+ * Returns the values that trigger autoloading from the options table.
+ *
+ * @since 6.6.0
+ *
+ * @return array The values that trigger autoloading.
+ */
+function wp_autoload_values_to_autoload() {
+	return array( 'yes', 'on', 'auto-on', 'auto' );
```

Comment:
See my feedback in https://github.com/WordPress/wordpress-develop/pull/5671#discussion_r1516639379, I think the filter you had there should be maintained but only to remove entries.
```suggestion
	$autoload_values = array( 'yes', 'on', 'auto-on', 'auto' );

	/**
	 * Filters the autoload values that should be considered for autoloading from the options table.
	 *
	 * The filter can only be used to remove autoload values from the default list.
	 *
	 * @since 6.6.0
	 *
	 * @param array $autoload_values Autoload values used to autoload option.
	 *                        Default list contains 'yes', 'on', 'auto-on', and 'auto'.
	 */
	$filtered_values = apply_filters( 'wp_autoload_values_to_autoload', $autoload_values );

	return array_intersect( $filtered_values, $autoload_values );
```